src/renderer/atlas/common.h

```diff
@@ -0,0 +1,496 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <dwrite_3.h>
+#include <dxgi1_2.h>
+
+#include <til/generational.h>
+
+#include "../../renderer/inc/IRenderEngine.hpp"
+
+namespace Microsoft::Console::Render::Atlas
+{
+#define ATLAS_FLAG_OPS(type, underlying)                                                       \

```

Comment:
All of this, down past the `Buffer` class, is existing code copied over from AtlasEngine.h.